assets/js/settings.ts

```diff
@@ -28,17 +28,41 @@ function setupThemeSettings() {
   themeColorSelect.addEventListener('change', themePreviewCallback);
 }
 
+function hideIf(element: HTMLElement, condition: boolean) {
+  if (condition) {
+    element.classList.add('hidden');
+  } else {
+    element.classList.remove('hidden');
+  }
+}
+
+function setupAutocompleteSettings() {
+  const autocompleteSettings = assertNotNull($<HTMLElement>('.autocomplete-settings'));
```

Comment:
Pick all the elements you need as early as you can

```suggestion
  const autocompleteSettings = assertNotNull($<HTMLElement>('.autocomplete-settings'));
  const enableSearchAutocomplete = assertNotNull($('#user_enable_search_ac'));
  const userSearchHistoryHidden = assertNotNull($('#user_autocomplete_search_history_hidden'));
```
```

Reply:
Applied the suggestion and refactored a bit